core/src/main/java/io/questdb/griffin/SqlExecutionContextImpl.java

```diff
@@ -121,6 +135,28 @@ public SqlExecutionInterruptor getSqlExecutionInterruptor() {
         return interruptor;
     }
 
+    @Override
+    public void storeTelemetry(short event) {
+        telemetryMethod.store(event);
+    }
+
+    private void doStoreTelemetry(short event) {
+        final long cursor = telemetryPubSeq.next();
+        TelemetryTask row = telemetryQueue.get(cursor);
```

Comment:
you absolutely must check the value of the cursor here. When queue is full, cursor == -1, the store has to be a noop. when cursor = -2, next() must be retried

Reply:
Done